### PR TITLE
Updated pkg to most recent version and updated Windows contributing instructions

### PR DESCRIPTION
**Pull request by ShadowCat567**
#### Description of changes
Updated Windows CONTRIBUTING.md instructions and updated pkg-fetch. pkg-fetch was creating windows build issues in its current form.
Workarounds related to pkg-fetch, no longer needed.
Fixed error in precommit hook for Windows if no files included in commit that match js/tsx extensions were modified.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/build_windows.yml [hunk 1]
```diff
@@ -4,10 +4,6 @@ env:
 phases:
   build:
     commands:
-      - cd scripts
-      - yarn
```
Comment (awsluja) [inline codebuild_specs/build_windows.yml:8]: might need to add these back in (lines 7, 8); appears that the scripts cache is missing in subsequent steps

Comment (awsluja) [inline codebuild_specs/build_windows.yml:8]: looks like build is failing and is expecting the scripts directory to have yarn install run there
image
It produces the scripts cache

Comment (ShadowCat567) [reply codebuild_specs/build_windows.yml:8]: ok, does that mean these lines should be re-instated?

Comment (awsluja) [reply codebuild_specs/build_windows.yml:8]: yes for lines 7 & 8, here, and in the other file where we also do cd scripts/yarn

Comment (ShadowCat567) [reply codebuild_specs/build_windows.yml:8]: Should I also add cd .. after yarn for this script?

Comment (ShadowCat567) [reply codebuild_specs/build_windows.yml:8]: updated build_windows.yml and action.yml to reinstate code that runs yarn install in the scripts file

Comment (ShadowCat567) [reply codebuild_specs/build_windows.yml:8]: it has been added

**Review by awsluja — APPROVED** [review]

**Review by 0618 — APPROVED** [review]
Finally, we get more obssessed to the Windows customers 🎉

**Pull request merged**